game/templates/game/stats.html

```
@@ -542,6 +565,45 @@ <h2>Opening Trainer Analytics</h2>
                 </div>
             </div>
         </div>
+        
+        <h2>Rating Progress Chart</h2>
+        {% if history_all %}
+        <div class="card" style="margin-top: 20px; text-align: left; padding: 24px;">
+            <div style="display: flex; justify-content: space-between; align-items: center; margin-bottom: 20px; flex-wrap: wrap; gap: 10px;">
+                <div style="font-size: 0.95rem; color: #94a3b8; font-weight: 600;">Rating Over Time</div>
+                <div class="timeframe-buttons" style="display: flex; gap: 8px;">
+                    <button class="btn-timeframe active" data-days="7">7 Days</button>
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)